lib/Model/ScheduledMessageMapper.php

```diff
@@ -0,0 +1,135 @@
+ * @method list<ScheduledMessage> findEntities(IQueryBuilder $query)
+ * @method ScheduledMessage update(ScheduledMessage $scheduledMessage)
+ * @method ScheduledMessage delete(ScheduledMessage $scheduledMessage)
+ * @template-extends QBMapper<ScheduledMessage>
+ */
+class ScheduledMessageMapper extends QBMapper {
+	use TTransactional;
+
+	public function __construct(
+		IDBConnection $db,
+	) {
+		parent::__construct($db, 'talk_scheduled_msg', ScheduledMessage::class);
+	}
+
+	/**
+	 * @throws DoesNotExistException
+	 */
+	public function findById(Room $chat, int $id, string $actorType, string $actorId): ScheduledMessage {
+		if (!$chat->isFederatedConversation()) {
+			throw new InvalidRoomException('Can not call ProxyCacheMessageMapper::findById() with a non-federated chat.');
+		}
+
+		$query = $this->db->getQueryBuilder();
+		$query->select('*')
+			->from($this->getTableName())
+			->where($query->expr()->eq('id', $query->createNamedParameter($id, IQueryBuilder::PARAM_INT)))
+			->andWhere($query->expr()->eq('actor_type', $query->createNamedParameter($actorType, IQueryBuilder::PARAM_STR)))
+			->andWhere($query->expr()->eq('actor_id', $query->createNamedParameter($actorId, IQueryBuilder::PARAM_STR)))
+			->andWhere($query->expr()->eq('room_id', $query->createNamedParameter($chat->getId(), IQueryBuilder::PARAM_STR)));
+
+		return $this->findEntity($query);
+	}
+
+	public function findByRoomAndActor(Room $chat, string $actorType, string $actorId): array {
+		$query = $this->db->getQueryBuilder();
+		$query->select('s.*');
+		$helper = new SelectHelper();
+		$helper->selectThreadsTable($query);
+		$helper->selectCommentsTable($query);
+		$query->from($this->getTableName(), 's')
+			->where($query->expr()->eq('s.room_id', $query->createNamedParameter($chat->getId(), IQueryBuilder::PARAM_STR)))
+			->andWhere($query->expr()->eq('s.actor_type', $query->createNamedParameter($actorType, IQueryBuilder::PARAM_STR)))
+			->andWhere($query->expr()->eq('s.actor_id', $query->createNamedParameter($actorId, IQueryBuilder::PARAM_STR)))
+			->leftJoin('s', 'talk_threads', 'th', $query->expr()->eq('s.thread_id', 'th.id'))
+			->leftJoin('s', 'comments', 'c', $query->expr()->eq('s.parent_id', 'c.id'));
+
+		$cursor = $query->executeQuery();
+		$result = $cursor->fetchAll();
+		$cursor->closeCursor();
+
+		return $result;
+	}
+
+	public function getCountByActorAndRoom(Room $chat, string $actorType, string $actorId): int {
+		$query = $this->db->getQueryBuilder();
+		$query->select(['room_id', $query->createFunction('COUNT(DISTINCT(room_id)) AS count'), 'room_id'])
+			->from($this->getTableName())
+			->where($query->expr()->eq('actor_type', $query->createNamedParameter($actorType, IQueryBuilder::PARAM_STR)))
+			->andWhere($query->expr()->eq('actor_id', $query->createNamedParameter($actorId, IQueryBuilder::PARAM_STR)))
+			->andWhere($query->expr()->eq('room_id', $query->createNamedParameter($chat->getId(), IQueryBuilder::PARAM_STR)));
```

Comment:
room_id was selected twice:
```suggestion
		$query->select(['room_id', $query->func()->count('*')])
			->from($this->getTableName())
			->where($query->expr()->eq('actor_type', $query->createNamedParameter($actorType, IQueryBuilder::PARAM_STR)))
			->andWhere($query->expr()->eq('actor_id', $query->createNamedParameter($actorId, IQueryBuilder::PARAM_STR)))
			->andWhere($query->expr()->eq('room_id', $query->createNamedParameter($chat->getId(), IQueryBuilder::PARAM_STR)))
			->groupBy('room_id');
```